### ISSUE TITLE
Java 17 NoSuchMethod Exception in RegExFilter

### ISSUE THREAD

**Issue by SamTV12345** [gwtproject/gwt#9792]
I am using GWT Version 2.10.0 with the GWT Maven plugin. During compilation I get a java.lang.NoSuchMethodError with stacktrace:
GTW Maven 2.10.0 is used for compilation.

[ERROR] Exception in thread "main" java.lang.NoSuchMethodError: com.google.gwt.util.regexfilter.RegexFilter: method 'void <init>()' not found
[ERROR]         at com.google.gwt.util.regexfilter.WhitelistRegexFilter.<init>(WhitelistRegexFilter.java:21)
[ERROR]         at com.google.gwt.dev.jjs.JJSOptionsImpl.<init>(JJSOptionsImpl.java:54)
[ERROR]         at com.google.gwt.dev.PrecompileTaskOptionsImpl.<init>(PrecompileTaskOptionsImpl.java:38)
[ERROR]         at com.google.gwt.dev.CompilerOptionsImpl.<init>(CompilerOptionsImpl.java:30)
[ERROR]         at com.google.gwt.dev.Compiler.main(Compiler.java:105)

**Comment by jnehlmeier** [comment]
First check your dependencies in your project and make sure there is no old GWT on classpath.

**Comment by SamTV12345** [comment]
Do you know which gwt version could be outdated? 

In my research I found the library  de.knightsoft-net which uses org.gwtproject.regexp:gwt-regexp:jar:1.0.0-RC1 . Could that be the problem?

**Comment by tbroyer** [comment]
From the error, we're talking about a version of GWT (gwt-dev or gwt-user) whose RegexFilter class would be missing a zero-argument constructor.
That constructor was added by this commit 9eda5dc418f9bdfef2fd21c6113e48dd10672beb, which shipped in 2.8.1; so check if you have any gwt-user (or gwt-dev) JAR in classpath from an earlier version (2.8.0 or earlier).
Actually, because some libs sometimes copied classes from GWT, look for any JAR containing a `com.google.gwt.util.regexfilter.RegexFilter` class.

**Comment by SamTV12345** [comment]
Thanks for the help. I found out that com.sencha.gxt Themebuilder copied the old file to their project. How can I prevent that this class is used?

**Comment by SamTV12345** [comment]
I fixed it with this method: https://gist.github.com/afrische/aa5cad210b637e2b5291 . I just copy `the com.google.gwt.util.regexfilter.RegexFilter` from the gwt-dev project to the root of /classes. That way this is the first class to be analyzed and taken into consideration.

**Comment by SamTV12345** [comment]
So in summary I used mvn dependency:tree to show the dependencies used in the project. I opened the file in intellij so I could go over every dependency including gwt. Because the Sencha Themebuilder copied the class and large parts of the code to their project this was not listened. After further research I clicked with STRG + Click on the RegExFilter class. In IntelliJ the dependency is shown when hovering over the class name. After that I added the fix above so the latest RegExFilter is the first to be analyzed. Thanks for the help :D .
